File: NetFramework/Ribbon/VB/WordPad/BackstageTabs/Items/ListItem.vb
Public Class ListItem

    Private _defaultBackColor As Color = Color.FromArgb(243, 243, 243)
    Private _hotBackColor As Color = Color.FromArgb(198, 198, 198)
    Private _hotPressedBackColor As Color = Color.FromArgb(195, 195, 195)

    Public Property Item As RecentDocumentItem
    Public Event PinnedChanged As EventHandler

    Public Sub OnPinnedChanged()
        RaiseEvent PinnedChanged(Me, EventArgs.Empty)
    End Sub

    Private Sub MenuItem_MouseDown(ByVal sender As Object, ByVal e As MouseEventArgs) Handles MyBase.MouseDown, pbPin.MouseDown, lblSubText.MouseDown, lblText.MouseDown, pbImage.MouseDown
        BackColor = _hotPressedBackColor
    End Sub

    Private Sub MenuItem_MouseUp(ByVal sender As Object, ByVal e As MouseEventArgs) Handles MyBase.MouseUp, pbPin.MouseUp, lblSubText.MouseUp, lblText.MouseUp, pbImage.MouseUp
        BackColor = _hotBackColor
    End Sub

    Private Sub MenuItem_MouseLeave(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.MouseLeave, pbPin.MouseLeave, lblSubText.MouseLeave, lblText.MouseLeave, pbImage.MouseLeave
        BackColor = _defaultBackColor
        UpdatePinnedState()
    End Sub

    Private Sub MenuItem_MouseEnter(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.MouseEnter, pbPin.MouseEnter, lblSubText.MouseEnter, lblText.MouseEnter, pbImage.MouseEnter
        BackColor = _hotBackColor
        UpdatePinnedState(True)
    End Sub

    Private Sub pbPin_Click(ByVal sender As Object, ByVal e As EventArgs) Handles pbPin.Click
        Item.Pinned = Not Item.Pinned
        OnPinnedChanged()
    End Sub

    Private Sub MenuItem_Click(ByVal sender As Object, ByVal e As EventArgs) Handles lblSubText.Click, lblText.Click, pbImage.Click
        OnClick(e)
    End Sub

    Public Sub SetItem(ByVal item As RecentDocumentItem, ByVal image As Image)
        Me.Item = item

        lblText.Text = item.Text
        lblSubText.Text = item.SubText
        pbImage.Image = image
        UpdatePinnedState()
    End Sub

    Private Sub UpdatePinnedState(ByVal Optional hot As Boolean = False)
        If Item.Pinned Then
            pbPin.Image = My.Resources.Pinnned_small
        ElseIf hot Then
            pbPin.Image = My.Resources.Pin_small
        Else
            pbPin.Image = Nothing
        End If

        pbPin.Invalidate()
    End Sub

    Private Sub ListItem_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        BackColor = _defaultBackColor
    End Sub
End Class

File: NetFramework/Ribbon/CS/WordPad/BackstageTabs/Items/ListItem.cs
using System;
using System.Drawing;
using System.Windows.Forms;

namespace WordPad.BackstageTabs.Items
{
    public partial class ListItem : UserControl
    {
        #region private fileds

        private Color _defaultBackColor = Color.FromArgb(243, 243, 243);
        private Color _hotBackColor = Color.FromArgb(198, 198, 198);
        private Color _hotPressedBackColor = Color.FromArgb(195, 195, 195);

        #endregion

        #region ctor

        public ListItem()
        {
            InitializeComponent();
            BackColor = _defaultBackColor;
        }

        #endregion

        #region properties

        public RecentDocumentItem Item { get; private set; }

        #endregion

        #region events

        public event EventHandler PinnedChanged;

        public void OnPinnedChanged() => PinnedChanged?.Invoke(this, new EventArgs());

        #endregion

        #region event handlers

        private void MenuItem_MouseDown(object sender, MouseEventArgs e)
        {
            BackColor = _hotPressedBackColor;
        }

        private void MenuItem_MouseUp(object sender, MouseEventArgs e)
        {
            BackColor = _hotBackColor;
        }

        private void MenuItem_MouseLeave(object sender, EventArgs e)
        {
            BackColor = _defaultBackColor;
            UpdatePinnedState();
        }

        private void MenuItem_MouseEnter(object sender, EventArgs e)
        {
            BackColor = _hotBackColor;
            UpdatePinnedState(true);
        }        

        private void pbPin_Click(object sender, EventArgs e)
        {
            Item.Pinned = !Item.Pinned;
            OnPinnedChanged();
        }

        private void MenuItem_Click(object sender, EventArgs e)
        {
            OnClick(e);
        }

        #endregion

        #region methods        

        public void SetItem(RecentDocumentItem item, Image image)
        {
            Item = item;
            lblText.Text = item.Text;
            lblSubText.Text = item.SubText;
            pbImage.Image = image;
            UpdatePinnedState();
        }

        private void UpdatePinnedState(bool hot = false)
        {
            if (Item.Pinned)
                pbPin.Image = Properties.Resources.Pinnned_small;
            else if (hot)
                pbPin.Image = Properties.Resources.Pin_small;
            else
                pbPin.Image = null;
            pbPin.Invalidate();
        }

        #endregion
    }
}
